### PR TITLE
deviseの導入

### LINKED ISSUE #97: deviseの導入
現在は、twitterのログイン機能があるが、メールログインを基本としたいので、deviseを導入する。

メールログインの流れについては別issueで対応。

http://qiita.com/cigalecigales/items/73d7bd7ec59a001ccd74

### PR DESCRIPTION
**Pull request by yh2020**
close #97

### CODE REVIEW

File: app/helpers/omniauth_callbacks_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module OmniauthCallbacksHelper
```
Comment (ms2sato) [inline app/helpers/omniauth_callbacks_helper.rb:1]: **[rails_best_practices]**
- remove empty helpers [:ref](http://rails-bestpractices.com/posts/2011/04/09/remove-empty-helpers/)

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users, controllers: { :omniauth_callbacks => "omniauth_callbacks" }
```
Comment (ms2sato) [inline config/routes.rb:2]: **[rubocop]**
- Style/HashSyntax: Use the new Ruby 1.9 hash syntax. [:ref](https://github.com/bbatsov/ruby-style-guide#hash-literals)

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, type: :controller do
+
```
Comment (ms2sato) [inline spec/controllers/omniauth_callbacks_controller_spec.rb:4]: **[rubocop]**
- Style/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.

**Comment by ms2sato** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://www.sideci.com/github_repositories/49850663/auto_correct/rubocop/pull_request/99/branch/97-add-devise)

See the documentation if you want to get more information about auto-correct.
[Document](https://www.sideci.com/ja/docs/autofix-rubocop)

File: spec/factories/users.rb [hunk 4]
```diff
@@ -4,5 +4,7 @@
     sequence(:uid) { |i| "uid#{i}" }
     sequence(:nickname) { |i| "nickname#{i}" }
     sequence(:image_url) { |i| "http://example.com/image#{i}.jpg" }
+    sequence(:email) { |i| "test#{i}@test.com" }
+    sequence(:password) { |i| "password" }
```
Comment (ms2sato) [inline spec/factories/users.rb:8]: **[rubocop]**
- Lint/UnusedBlockArgument: Unused block argument - `i`. You can omit the argument if you don't care about it. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

**Comment by yh2020** [comment]
Created Auto-correct Pull Request from [SideCI](https://www.sideci.com/)

[https://github.com/CircleAround/eventure/pull/106](https://github.com/CircleAround/eventure/pull/106)

**Pull request merged**